### PR TITLE
added mechanism to add s3 link in export csv

### PR DESCRIPTION
**Pull request by Shriyanshagro**

### CODE REVIEW

File: app/services/submissions/csv_export_service.rb [hunk 1]
```diff
@@ -26,6 +27,14 @@ def call
             submission.grading_status,
             submission.challenge_round&.challenge_round
           ]
+          if downloadable & submission.submission_files.present?
+            links = []
```
Comment (skbly7) [inline app/services/submissions/csv_export_service.rb:31]: consider empty case

File: app/services/submissions/csv_export_service.rb [hunk 2]
```diff
@@ -26,6 +27,14 @@ def call
             submission.grading_status,
             submission.challenge_round&.challenge_round
           ]
+          if downloadable & submission.submission_files.present?
```
Comment (skbly7) [inline app/services/submissions/csv_export_service.rb:30]: & -> &&

File: spec/services/submissions/csv_export_service_spec.rb [hunk 3]
```diff
@@ -1,25 +1,38 @@
 require 'rails_helper'
 
 describe Submissions::CSVExportService do
-  subject { described_class.new(submissions: Submission.all) }
 
   let!(:participant)     { create(:participant, name: 'Jhon') }
   let!(:challenge_round) { create(:challenge_round, score_title: 'Test Title', score_secondary_title: 'Test Secondary Title') }
   let!(:submissions)     { create_list(:submission, 3, participant: participant, challenge_round: challenge_round) }
```
Comment (skbly7) [inline spec/services/submissions/csv_export_service_spec.rb:7]: `:with_file` can be tried

**Review by skbly7 — CHANGES_REQUESTED** [review]

File: app/services/submissions/csv_export_service.rb (hunk 1 above)
Comment (Shriyanshagro) [reply app/services/submissions/csv_export_service.rb:31]: done

File: app/services/submissions/csv_export_service.rb (hunk 2 above)
Comment (Shriyanshagro) [reply app/services/submissions/csv_export_service.rb:30]: not required anymore

File: spec/services/submissions/csv_export_service_spec.rb (hunk 3 above)
Comment (Shriyanshagro) [reply spec/services/submissions/csv_export_service_spec.rb:7]: done

File: app/services/base_service.rb [hunk 4]
```diff
@@ -18,4 +18,14 @@ def success(value = true)
   def failure(value = false)
     Result.new(false, value)
   end
+
+  def s3_expiring_url(s3_key)
```
[the remark this answers is not in the document]
Comment (Shriyanshagro) [reply app/services/base_service.rb:22]: resolved

**Review by skbly7 — APPROVED** [review]

**Pull request merged**